src/onConsentChange.test.js

```diff
@@ -51,4 +60,22 @@ describe('onConsentChange', () => {
 				}),
 			);
 	});
+
+	it('returns all 10 TCF purposes even if they are false', () => {
+		const consents = {
+			'1': false,
+			'2': false,
+			'3': false,
+			'4': false,
+			'5': false,
+			'6': false,
+			'7': false,
+			'8': false,
+			'9': false,
+			'10': false,
+			...tcfData.purposes.consents,
+		};
+
+		expect(Object.keys(consents)).toHaveLength(10);
+	});
```

Comment:
Maybe few more expectations will help
```suggestion
	expect(consents['1']).toBe(true);
	expect(consents['2']).toBe(false);
	expect(consents['3']).toBe(true);
	});
```

Reply:
Technically, this should also be using a "real" test. I did not manage to mock the `__tcfapi` properly, however.

Reply:
I've moved this into its own function call and tested that instead!

Reply:
Nice one !  👍